src/lib.rs

```diff
@@ -1277,6 +1277,44 @@ impl Error {
         self.depth
     }
 
+    /// Inspect the underlying [`io::Error`] if there is one.
+    /// If the underlying error isn't an [`io::Error`], [`None`] is returned.
```

Comment:
Could the documentation be expanded to mention when this might not contain an `io::Error`? e.g., "In some circumstances, this `Error` won't correspond to an `io::Error`, for example, when the error was produced because a cycle was found in the directory tree while following symbolic links."

Reply:
I think the docs should also mention that if the caller wants to acquire an owned `io::Error`, then they can convert this error into an `io::Error` using `io::Error::from(err)`. And state that if this error doesn't correspond to an `io::Error`, then one is created.